### ISSUE TITLE
As a registered user, I want to be sure OTP is properly configured before it is activated

### ISSUE THREAD

**Issue by szpak** [fedora-infra/noggin#422]
### Sample scenario for registered user

1. Log in
2. Enable OTP for account
3. Scan QR code and (seem to) configure some app for OTP

### Problematic corner case

4. Close the OTP dialog and log out
5. Some time later try to log in to notice that the OTP app is not properly configured (the token just doesn't work - bug in application? corrupted QR scan? browser crashed?).
6. Account is locked :cry: .

### Corner case mitigation

4. User is asked to enter valid open-time password (in the same dialog to scan QR code).
5. (Only after successful validation) OTP is permanently enabled for the user.
6. OTP dialog is automatically closed only after OTP verification (user can also cancel the whole operation before entering QR code).

**Comment by szpak** [comment]
Thanks @abompard! Btw, when it will be ready for test on https://accounts.stg.centos.org/ ?

**Comment by abompard** [comment]
It should be deployed now! Please tell me if it does not work for you.

**Comment by szpak** [comment]
It works like a harm!

Just some nitpicking. The QR code is revealed by default (which is very sensible), however, there is still a button "Reveal" above it:
![image](https://user-images.githubusercontent.com/148013/114936383-a8135e00-9e3c-11eb-9578-3ff7b7022e1e.png)

It wonder, if an option to conceal the image is still needed? If yes, maybe that button could change its name depending on the state (Conceal/Hide is visible and Reveal/Show otherwise)?

**Comment by szpak** [comment]
Once more things. The "Verification code" edit field in that modal dialog keeps history of entered values (the same as regular edit fields). I don't think it makes sense to suggest old OTP codes every time you try to enter new one (e.g. for the another token) :-).

**Comment by abompard** [comment]
> Just some nitpicking. The QR code is revealed by default (which is very sensible), however, there is still a button "Reveal" above it:

Yeah that's #577 .

> Once more things. The "Verification code" edit field in that modal dialog keeps history of entered values (the same as regular edit fields). I don't think it makes sense to suggest old OTP codes every time you try to enter new one (e.g. for the another token) :-).

Yep, I didn't bother changing the default behavior of WTForms. Please open a separate ticket if you find it annoying :-)

Thanks for the feedback!

**Comment by szpak** [comment]
I generated a few more token and there is a change what identification data are put into the QR code/URL. In the previous version, I had (my own) `token name` (as name) and `login@stg.fedoraproject.org` (as issuer). In the new version there are `my user name` and `token name` (reversed).
> otp_uri = token.provisioning_uri(name=username, issuer_name=description)

I somehow miss the service name put in the issuer field. In your OTP application you usually (assuming production use) have just one entry for a given server. The token name is rather useful to distinguish it in the service when you might have more token (e.g. for your personal Android phone and your work iPhone) - to know which is which when you want to revoke/remove one. In addition, when I didn't provide a token name (why not, having just one token in fedoraproject.org) my OTP client (Aegis) didn't show any label related to my token (I needed to add it manually).

Looking at the "entries" generated by other services, the service name is rather used as a issuer. What do you think about bringing it back for clarity? :-)

(I might create a separate issue, if you like the idea)
